### ISSUE TITLE
Adding option to not treat digits as word boundaries

### ISSUE THREAD

**Issue by savvasdalkitsis** [pearxteam/kasechange#6]
The latest version of the library changed the behaviour when it comes to how digits are treated (commit https://github.com/pearxteam/kasechange/commit/e8e3222745956738aa8c584d9fc199ea2b37cb77).

It used to be that something like `"a_word1_with_number" == "a_word1_with_number".toSnakeCase()` but it no longer holds true.

To allow clients to maintain that behaviour if they need to so that they can still update to later versions of the library (my project is one that needs this behaviour for various reasons but I also would like to use the latest version), I raised https://github.com/pearxteam/kasechange/pull/5

This PR maintains API compatibility as it introduces one parameter to the public API that has a default value so existing code should not be affected

**Comment by mrapplexz** [comment]
I'm planning to rewrite the library to make it even more configurable, so this feature will be included in the next update

**Comment by stellingsimon** [comment]
This issue results in very surprising behaviour:

```
"f766e60e-4ea2-4169-bd0c-e24f4c3848bf".toKebabCase() == "f-76-6e-6-0e-4ea-2-4169-bd-0c-e-2-4f-4c-384-8bf" 
```

I would have considered `"f766e60e-4ea2-4169-bd0c-e24f4c3848bf"` to be kebab-cased, so the `.toKebabCase()` should have returned the same value.

**Comment by mrapplexz** [comment]
I've released kasechange 1.2.0. It allows you to specify a WordSplitter in case-converting functions.
For example, you can use one of the `CaseFormat`s to convert string from specific case or use universal `WordSplitter`:
```kotlin
"a_word1_with_number".toSnakeCase(CaseFormat.LOWER_UNDERSCORE) // a_word1_with_number
"a_word1_with_number".toSnakeCase(universalWordSplitter(treatDigitsAsUppercase = false)) // a_word1_with_number
```
